ocaml/xapi/xapi_host.ml

```diff
@@ -3074,23 +3074,25 @@ let apply_recommended_guidances ~__context ~self:host =
   Helpers.assert_we_are_master ~__context ;
   try
     let open Updateinfo in
-    Db.Host.get_recommended_guidances ~__context ~self:host |> function
+    Db.Host.get_pending_guidances ~__context ~self:host
+    (* Ingore the guidance as this has to be handled by user *)
+    |> List.filter (fun g -> g <> `reboot_host_on_livepatch_failure)
+    |> function
     | [] ->
-        try_restart_device_models_for_recommended_guidances ~__context ~host
+        try_restart_device_models ~__context ~host
     | [`reboot_host] ->
         Helpers.call_api_functions ~__context (fun rpc session_id ->
             Client.Client.Host.reboot ~rpc ~session_id ~host
         )
     | [`restart_toolstack] ->
-        try_restart_device_models_for_recommended_guidances ~__context ~host ;
+        try_restart_device_models ~__context ~host ;
```

Comment:
Why do you restart the device model when restarting the toolstack? Aren't they 2 separate guidances that are orthogonal?
(I assume restart device model = restart qemu by migrating the VM?)

Reply:
We'll address this separately (it is not caused by this PR).